### DISCUSSION TITLE
How do I make a hamburger menu?

### DISCUSSION THREAD

**Discussion by glow12121** [datenstrom/community#651] [Ask a question] [answered]
Hi,

Is there an easy way to make a hamburger menu for a Yellow-generated page, if display space is too small  for a regular navigation menu. Something in the way Github does with its menu.

Thanks.

**Comment by nzyeehaa** [comment]
sounds like you need a web designer, but for learning purposes - have a look here datenstrom/community#213 

https://github.com/richi/yellow-plugin-burger-menu

**Reply by glow12121** [reply]
@nzyeehaa thanks, that's probably what I need. Will try the plugin out later.

I don't want to hire a designer because I don't take my site that seriously for now.

**Reply by schulle4u** [reply]
**Note**: This extension currently is not compatible with latest Yellow API and probably will break your site. I suggest to file an issue to @richi if he still maintains this project and could update the extension.

**Reply by glow12121** [reply]
Thanks, @schulle4u, I've done this.

**Comment by GiovanniSalmeri** [comment]
Hello @thumos333 This site is made with Yellow: https://mondodomani.org/filosofiatorvergata/ Do you mean a hamburger menu like this? I could explain how I made it

**Comment by glow12121** [comment]
Yes, please do.

**Comment by IngoBelka** [comment]
I'm was looking for a solution since a long time, too. For that I played around with W3Schools:
https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_topnav
For my tries I took the navigation of my page. 
This is my current state and not the goal yet:
```
<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<style>
body {
  margin: 0;
  font-family: Arial, Helvetica, sans-serif;
}

ul {
  padding: 0;
  margin: 0;
  border: 0;
}

.navigation {
  overflow: hidden;
  background-color: datenstrom/community#333;
}

.navigation li {
  float: left;
  display: block;
}
.navigation li a {
  display: block;
  color: #f2f2f2;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
  font-size: 17px;
}

.navigation li a:hover {
  background-color: #ddd;
  color: black;
}

.navigation li a.active {
  background-color: #04AA6D;
  color: white;
}

.navigation li.icon {
  display: none;
}

@media screen and (max-width: 600px) {
  .navigation li {display: none;}
  .navigation li.icon {
    float: right;
    display: block;
  }
}

@media screen and (max-width: 600px) {
  .navigation.responsive {position: relative;}
  .navigation.responsive li.icon {
    position: absolute;
    right: 0;
    top: 0;
  }
  .navigation.responsive li {
    float: none;
    display: block;
}
  .navigation.responsive li a {

    text-align: left;
  }
}
</style>
</head>
<body>

<div class="navigation" role="navigation" id="navigation">
<ul>
<li><a class="active" aria-current="page" href="/kristall/">Start</a></li>
<li><a href="/kristall/das-bündnis/">Das Bündnis</a></li>
<li><a href="/kristall/die-partner/">Die Partner</a></li>
<li><a href="/kristall/blog/">Neuestes</a></li>
<li><a href="/kristall/Impressum/">Impressum</a></li>
<li class="icon">
    <a href="javascript:void(0);" class="icon" onclick="myFunction()">
    <i class="fa fa-bars"></i>
    </a>
</li>  
</ul>
</div>

<div style="padding-left:16px">
  <h2>Responsive Topnav Example</h2>
  <p>Resize the browser window to see how it works.</p>
</div>

<script>
function myFunction() {
  var x = document.getElementById("navigation");
  if (x.className === "navigation") {
    x.className += " responsive";
  } else {
    x.className = "navigation";
  }
}
</script>

</body>
</html>
```
You can take this code to try yourself in W3Schools.
I stuck now, because I don't know, how to get this into Yellow.

What do you think about it? It's certainly nothing special, but one have to start somewhere.

**Comment by GiovanniSalmeri** [comment]
@thumos333 I have published an extension https://github.com/GiovanniSalmeri/yellow-burger

It is not an entire layout, I think that it can be easily used with different layouts. <del>As soon as possible I will made a screenshot, but the final result is as in the site https://mondodomani.org/filosofiatorvergata/</del> Comments are welcome!

**Reply by IngoBelka** [reply]
@GiovanniSalmeri  I tried your extension mini-burger, because it fits my needs most. Thank for your efforts. It works fine. Now I have to set up my own css-settings. Let's see, what I do get in the end.

**Reply by GiovanniSalmeri** [reply]
You are welcome, I'm glad I could help!

**Reply by satori-books** [reply]
@GiovanniSalmeri probably this is not the point here but is it possible to adapt this useful extension to show the 'tags' ?

**Reply by GiovanniSalmeri** [reply]
@satori-books I didn't ever think about this possibility, but probably it is easy to achieve this. (Anyway, are you speaking about [miniburger](https://github.com/GiovanniSalmeri/yellow-miniburger) or about [burger](https://github.com/GiovanniSalmeri/yellow-burger)? They are rather different beasts.)

**Reply by satori-books** [reply]
I was speaking about the [burger](https://github.com/GiovanniSalmeri/yellow-burger) : )  like having a few pages + subpages followed by the list of 'tags', but a 'only tags' version would be also highly appreciated ..

**Reply by satori-books** [reply]
@GiovanniSalmeri I've tried to insert in  the _burger.html_ different solutions proposed in [#842](https://github.com/datenstrom/yellow/discussions/842) but nothing works.. That said I have to admit to be a total noobie with code/php etc.

**Comment by glow12121** [comment, marked as answer]
@IngoBelka, @GiovanniSalmeri, thank you for your ideas. In my view, the simplest and best burger menu is something along the lines of what GitHub has. A menu that drops down upon tap and moves the rest of the page down. It is the most good-looking and least glitchy option, from my experience. It is something like what the W3 School suggests, too.

**Reply by GiovanniSalmeri** [reply]
@thumos333 Do you mean the first example at page https://www.w3schools.com/howto/howto_js_mobile_navbar.asp ? If so, it is seems rather easy to adapt it to Yellow. Just allow me a little of time... 😊

**Reply by glow12121** [reply]
This example looks nice, in my view:

![Screenshot from 2021-05-02 19-32-42](https://user-images.githubusercontent.com/29035180/116820365-56c5d700-ab7d-11eb-81be-8bc2b7b25c9f.png)

![Screenshot from 2021-05-02 19-33-00](https://user-images.githubusercontent.com/29035180/116820366-588f9a80-ab7d-11eb-9495-5f5544bcbcba.png)

**Reply by IngoBelka** [reply]
@GiovanniSalmeri  I'm looking foward to see your solution. I like it as easy as possible. Thank you in advance for your efforts.

**Reply by GiovanniSalmeri** [reply]
Here you are: https://github.com/GiovanniSalmeri/yellow-miniburger

I tried to modify the solution of W3Schools as little as possible. In `miniburger.css` you will find the stylesheet (identical, I just added a little tweak); in `miniburger.js` a slightly modified js code (Yellow does not allow the attribute `onclick`, therefore it must be dinamically added); in `miniburger.php` the little boilerplate; in `miniburger.html` a variation of `navigation.html` which create the menu with the format expected by the stylesheet. It's not perfect, but in simple cases I hope it will be useful!

**Reply by glow12121** [reply]
Many thanks for your work, @GiovanniSalmeri. I've noticed that the extension creates an additional folder for layouts. `Layouts`, in the plural, is the original one:

![Screenshot from 2021-05-05 21-01-42](https://user-images.githubusercontent.com/29035180/117188125-60ea0e80-ade5-11eb-837d-f95b5f34bd61.png)

Otherwise, the extension works just fine for my site!

**Reply by markseuffert** [reply]
Good work @GiovanniSalmeri. I would recommend to use the [file actions](https://github.com/datenstrom/yellow-extensions/blob/master/source/publish/README.md#settings-actions) `create, update, careful` for layout files. Then people can customise layouts, without them being accidentally overwritten in an update.

**Reply by GiovanniSalmeri** [reply]
@markseu @thumos333 Thank you very much, fixed.

**Reply by glow12121** [reply]
@GiovanniSalmeri, thank you.

**Comment by JetGum** [comment]
This is great, love its simplicity. Any plans to adapt this for navigation-tree?
Cheers!

Carsten

**Reply by GiovanniSalmeri** [reply]
Well, more than plans there are desires :-) Navigation is one of the key aspect of usability, so efforts would be well spent, but just now I am in great shortage of time... But I have written this in my wish-list!

**Reply by glow12121** [reply]
I actually had a suggestion to update all default Yellow templates so they include a hamburger menu in navigation on smaller screens. I could contribute simple pictures explicating the idea.
